src/passes/pass.cpp

```diff
@@ -138,19 +138,23 @@ void PassRunner::addDefaultGlobalOptimizationPasses() {
 }
 
 void PassRunner::run() {
-  if (options.debug) {
+  // BINARYEN_PASS_DEBUG is a convenient commandline way to log out the toplevel passes, their times,
+  //                     and validate between each pass.
+  //                     (we don't recurse pass debug into sub-passes, as it doesn't help anyhow and
+  //                     also is bad for e.g. printing which is a pass)
+  static const int passDebug = getenv("BINARYEN_PASS_DEBUG") ? atoi(getenv("BINARYEN_PASS_DEBUG")) : 0;
+  if (!isNested() && (options.debug || passDebug)) {
```

Comment:
I see it could work that way too, but I think we do still want `isNested()`, as it should be useful elsewhere. An example could be that nested passes do not want to use multiple threads, or have other limits on their parallelization. They may also want to validate differently.

Splitting up run into doRun and doRunDebug seems like a separate issue. No strong feelings on that, if you think it's important to do here I can do it.